src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -972,38 +949,28 @@ void handle_worker_available(LocalSchedulerState *state,
                              LocalSchedulerClient *worker) {
   CHECK(worker->task_in_progress == NULL);
   /* Check that the worker isn't in the pool of available workers. */
-  for (LocalSchedulerClient **p = (LocalSchedulerClient **) utarray_front(
-           algorithm_state->available_workers);
-       p != NULL; p = (LocalSchedulerClient **) utarray_next(
-                      algorithm_state->available_workers, p)) {
-    DCHECK(*p != worker);
-  }
+  CHECK(!worker_in_vector(algorithm_state->available_workers, worker));
+
   /* Check that the worker isn't in the list of blocked workers. */
-  for (LocalSchedulerClient **p = (LocalSchedulerClient **) utarray_front(
-           algorithm_state->blocked_workers);
-       p != NULL; p = (LocalSchedulerClient **) utarray_next(
-                      algorithm_state->blocked_workers, p)) {
-    DCHECK(*p != worker);
-  }
+  CHECK(!worker_in_vector(algorithm_state->blocked_workers, worker));
+
   /* If the worker was executing a task, it must have finished, so remove it
-   * from the list of executing workers. */
-  for (int i = 0; i < utarray_len(algorithm_state->executing_workers); ++i) {
-    LocalSchedulerClient **p = (LocalSchedulerClient **) utarray_eltptr(
-        algorithm_state->executing_workers, i);
-    if (*p == worker) {
-      utarray_erase(algorithm_state->executing_workers, i, 1);
-      break;
-    }
-  }
-  /* Check that we actually erased the worker. */
-  for (int i = 0; i < utarray_len(algorithm_state->executing_workers); ++i) {
-    LocalSchedulerClient **p = (LocalSchedulerClient **) utarray_eltptr(
-        algorithm_state->executing_workers, i);
-    DCHECK(*p != worker);
+   * from the list of executing workers. If the worker is connecting for the
+   * first time, it will not be in the list of executing workers. */
+  auto it = std::find(algorithm_state->executing_workers.begin(),
+                      algorithm_state->executing_workers.end(), worker);
+  if (it != algorithm_state->executing_workers.end()) {
+    std::swap(*it, algorithm_state->executing_workers.back());
```

Comment:
Same here for if the pointers are the same. It would be good to have a helper function for this.